kotlinpoet/src/main/java/com/squareup/kotlinpoet/PropertySpec.kt

```diff
@@ -54,6 +58,10 @@ public class PropertySpec private constructor(
     require(mutable || setter == null) {
       "only a mutable property can have a setter"
     }
+    if (contextReceiverTypes.isNotEmpty()) {
+      check(!mutable) { "mutable properties cannot have context receivers" }
```

Comment:
Mutable properties _can_ have context receivers, the snippet below compiles:

```kotlin
context(LoggingContext)
var foo: String
  get() = ""
  set(value) {}
```

The rule here seems to be:

- If the property is immutable - it has to have a custom getter
- If the property is mutable - it has to have custom accessors

Also, let's use `require` everywhere for consistency (unless there's a specific reason to use `check`?).